extension/app/ts/components/subcomponents/address.tsx

```diff
@@ -128,11 +105,15 @@ export function ActiveAddress(params: ActiveAddressParams) {
 		</div>
 
 		<div class = 'media-content' style = 'overflow-y: hidden;'>
-			<CopyToClipboard content = { ethers.utils.getAddress(addressString(params.address)) } copyMessage = 'Address copied!'>
-				<p class = 'title is-5 noselect nopointer' style = 'text-overflow: ellipsis; white-space: nowrap;'>
-					{ title }
-				</p>
-			</CopyToClipboard>
+			<div style = 'display: flex; position: relative;'>
+				<RenameAddressButton renameAddress = { () => params.renameAddressCallBack(title, addressString(params.address)) }>
+					<CopyToClipboard content = { ethers.utils.getAddress(addressString(params.address)) } copyMessage = 'Address copied!'>
+						<p class = 'title is-5 noselect nopointer' style = 'text-overflow: ellipsis; white-space: nowrap;'>
+							{ title }
+						</p>
+					</CopyToClipboard>
+				</RenameAddressButton>
```

Comment:
Why does the rename address button wrap the clipboard?  That feels very bizarre to me.  If you click the rename address button you wouldn't want it to copy the address I don't think.

Reply:
it doesn't copy the address in that case as `RenameAddressButton` drawn over the text hijacks the click event and doesn't forward it to the `CopyToClipboard `

Reply:
Why is `CopyToClipboard` a child of `RenameAddressButton` then?  It feels like the hierarchy should be something like:
```html
<Address>
	<CopyToClipboard>
		<AddressString/>
	</CopyToClipboard>
	<RenameAddressButton/>
</Address>
```

Reply:
`CopyToClipboard` a child of `RenameAddressButton` because its trigger is underneath, we first check mouseovers on the address string, then make `RenameAddressButton` visible and at the same time show the tooltip of `CopyToClipboard`. On Click of the element, we'll first check if `RenameAddressButton` is pressed, and if so, open the dialog, if not, we copy the address with `CopyToClipboard`.

I believe we could maybe use the hierarchy you suggested, but I don't really know how. Maybe using `ref` to get the hover effect trigger to `RenameAddressButton`?